### ISSUE TITLE
VideoFrame.to_ndarray with format "rgb48be"

### ISSUE THREAD

**Issue by pfk-beta** [PyAV-Org/PyAV#567]
First of all, I would like to thank you for wonderful library for manipulating videos in Python:)

## Overview
I have a HDR movie in `yuv420p10le` format, and I would like to extract some frames to `rgb48be`, which I do with following ffmpeg command: `ffmpeg -i <input-file> -pix_fmt rgb48be <some-other-options> frame_%06d.png`.

## Expected PyAV API
```python
frame.reformat(1920, 1080, 'rgb48be')
new_frame.to_ndarray()  # ValueError: Conversion to numpy array with format `yuv420p10le` is not yet supported
```
or
```python
np_array_of_yuv420p10le = frame.to_ndarray() # ValueError: Conversion to numpy array with format `yuv420p10le` is not yet supported
```

## Investigation
implement additional conversion in av/video/frame.pyx in VideoFrame::to_ndarray
--edit--
reimplement useful_array function in av/video/frame.pyx, to pass numpy's type

## Reproduction

pseudocode:
```
- generate with numpy/pillow data in yuv420p10le
- create frame with created already data
- convert frame to desired pix format
- test it
```

## Versions

- OS: docker
- PyAV runtime:
```
av.__version__
'6.2.0'
```

PS
how to you run tests?

**Comment by mikeboers** [comment]
I'm a bit confused by your exception message. If you were doing

```
new_frame = frame.reformat(1920, 1080, 'rgb48be')
my_array = new_frame.to_ndarray()
```

wouldn't the `ValueError` complain about `rgb48be` being missing?

I can imagine how we might implement the `rgb48` series, but I don't think we can do `yuv420p10le` because numpy doesn't work with not-byte-multiple integers.

**Comment by mikeboers** [comment]
Also, you run tests (once already in the environment via `source scripts/activate`) via `make test`, `./scripts/test` or `python -m unittest`. Whichever you feel like at the moment. I'm usually lazy and use `make test` cause it also builds it, and is less to type.

**Comment by pfk-beta** [comment]
> I'm a bit confused by your exception message. If you were doing
> 
> ```
> new_frame = frame.reformat(1920, 1080, 'rgb48be')
> my_array = new_frame.to_ndarray()
> ```
> 
> wouldn't the `ValueError` complain about `rgb48be` being missing?
> 
> I can imagine how we might implement the `rgb48` series, but I don't think we can do `yuv420p10le` because numpy doesn't work with not-byte-multiple integers.

copy-paste typo, `ValueError: Conversion to numpy array with format `rgb48be` is not yet supported`. I have solved issue outside of PyAV. So we may close it:) sorry for alarm

**Comment by mikeboers** [comment]
Cooooool.

I'm going to leave this open until I either add the feature or move it to a TODO list somewhere.

🍻

**Comment by jlaine** [comment]
Fixed in #939
